packages/react-file-tree/src/tree/Tree.js

```diff
@@ -101,17 +104,22 @@ export default class extends Component {
       <div style={styles.nodeContainer}
         tabIndex={'0'}
         onClick={this.handleClick.bind(this, node, nodeMetadata, index)}
+        onDoubleClick={this.handleDoubleClick.bind(this, node, nodeMetadata, index)}
+        onMouseEnter={this.handleMouseEnter.bind(this, node, nodeMetadata, index)}
+        onMouseLeave={this.handleMouseLeave.bind(this, node, nodeMetadata, index)}
         onKeyUp={this.handleKeyUp.bind(this, node, nodeMetadata, index)}
         onKeyDown={this.handleKeyDown.bind(this, node, nodeMetadata, index)}
         onContextMenu={this.handleContextMenu.bind(this, node, nodeMetadata, index)}
       >
-        <NodeComponent
-          key={path}
-          node={node}
-          metadata={nodeMetadata}
-          depth={depth}
-          index={index}
-        />
+        {
+          React.cloneElement(NodeComponent, {
```

Comment:
Ok, seems like an important use case. I haven't seen an API like what's in this PR before. Here's what I'm thinking. Instead of passing `NodeComponent`, we expose a `renderNode` prop. `renderNode={(props) => {}}` where props contains all the handlers, onClick, etc. And then you can spread them into the node.

```
renderNode={(props) => <Node foo='bar' {...props} />}
``` 

I can think of a few other potential APIs, but nothing I like better. Do you have any ideas?